src/test/java/org/gridsuite/useradmin/server/UserProfileTest.java

```diff
@@ -252,8 +289,16 @@ private UUID createProfile(String profileName, String userName, Integer maxAllow
             // check repository
             Optional<UserProfileEntity> profile1 = userProfileRepository.findByName(profileName);
             assertTrue(profile1.isPresent());
-            assertEquals(maxAllowedCases, profile1.get().getMaxAllowedCases());
-            assertEquals(maxAllowedBuilds, profile1.get().getMaxAllowedBuilds());
+            if (maxAllowedCases == null) {
+                assertEquals(userAdminApplicationProps.getDefaultMaxAllowedCases(), profile1.get().getMaxAllowedCases());
+            } else {
+                assertEquals(maxAllowedCases, profile1.get().getMaxAllowedCases());
+            }
+            if (maxAllowedBuilds == null) {
+                assertEquals(userAdminApplicationProps.getDefaultMaxAllowedBuilds(), profile1.get().getMaxAllowedBuilds());
+            } else {
+                assertEquals(maxAllowedBuilds, profile1.get().getMaxAllowedBuilds());
+            }
```

Comment:
```suggestion
            assertEquals(
                Optional.ofNullable(maxAllowedCases).orElse(userAdminApplicationProps.getDefaultMaxAllowedCases()),
                profile1.get().getMaxAllowedCases()
            );
            assertEquals(
                Optional.ofNullable(maxAllowedBuilds).orElse(userAdminApplicationProps.getDefaultMaxAllowedBuilds()),
                profile1.get().getMaxAllowedBuilds()
            );
```
You can use optionals here :grin: